app/models/recipe.rb

```diff
@@ -1,3 +1,10 @@
 class Recipe < ApplicationRecord
    validates_presence_of :name, :complexity, :genre
+
+   has_many :recipe_ingredients
+   has_many :ingredients, through: :recipe_ingredients
+
+   def total_cost
+      self.ingredients.sum(:cost)
```

Comment:
```suggestion
      ingredients.sum(:cost)
```
Instance methods will always be called on `self` implicitly so we don't need to say it explicitly. It's more conventional in ruby to leave it off if we don't need to say it.